bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/ProgressCallbackImpl.java

```diff
@@ -48,8 +49,16 @@
     private Collection<ProgressStep> sequence;
     private Iterator<ProgressStep> progressIterator;
     private ProgressStep current;
-    private boolean finished;
-    private boolean pending;
+    private int progress;
+
+    private enum InternalState {
```

Comment:
Otherwise I would need boolean flags for pending and finished. 

I think this enum makes the possible states of the ProgressCallback more visible and better to understand.  Furthermore it ensures that the callback is always is in a defined state and therefore is more failsafe.